apps/api/src/notifications/notifications.service.ts

```diff
@@ -0,0 +1,314 @@
+import { BadRequestException, Injectable, Logger } from '@nestjs/common'
+import { PrismaService } from '../prisma/prisma.service'
+import { SendConfirmationDto, SubscribePublicDto } from './dto/subscribe.dto'
+import { PersonService } from '../person/person.service'
+import { ConfirmConsentEmailDto } from '../email/template.interface'
+import { ConfigService } from '@nestjs/config'
+import { EmailService } from '../email/email.service'
+import {
+  EmailSentRegistry,
+  EmailType,
+  Person,
+  UnregisteredNotificationConsent,
+} from '@prisma/client'
+import { NotificationsProviderInterface } from './providers/notifications.interface.providers'
+import { SendGridParams } from './providers/notifications.sendgrid.types'
+import { DateTime } from 'luxon'
+import * as crypto from 'crypto'
+import { CampaignService } from '../campaign/campaign.service'
+import { KeycloakTokenParsed } from '../auth/keycloak'
+
+@Injectable()
+export class MarketingNotificationsService {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly personService: PersonService,
+    private readonly config: ConfigService,
+    private readonly sendEmail: EmailService,
+    private readonly campaignService: CampaignService,
+    private readonly marketingNotificationsProvider: NotificationsProviderInterface<SendGridParams>,
+  ) {}
+
+  async sendConfirmConsentEmail(data: SendConfirmationDto) {
```

Comment:
it looks like this function returns, a status object - could you please add a type for it?

Reply:
Done